Documentation/ceph-cluster-crd.md

```diff
@@ -444,6 +452,61 @@ You can set priority class names for Rook components for the list of key value p
 
 The specific component keys will act as overrides to `all`.
 
+### Health settings
+
+Rook-Ceph will monitor the state of the CephCluster on various components by default.
+The following CRD settings are available:
+
+* `healthCheck`: main ceph cluster health monitoring section
+
+Currently three health checks are implemented:
+
+* `mon`: health check on the ceph monitors, basically check whether monitors are members of the quorum. If after a certain timeout a given monitor has not joined the quorum back it will be failed over and replace by a new monitor.
+* `osd`: health check on the ceph osds
+* `status`: ceph health status check, periodically check the Ceph health state and reflects it in the CephCluster CR status field.
+
+The liveness probe of each daemon can also be controlled via `livenessProbe`, the setting is valid for `mon`, `mgr` and `osd`.
+Here is a complete example for both `daemonHealth` and `livenessProbe`:
+
+```yaml
+healthCheck:
+  daemonHealth:
+    mon:
+      disabled: false
+      interval: 60s
+    osd:
+      disabled: false
+      interval: 60s
+    status:
+      disabled: false
+  livenessProbe:
+    mon:
+      disabled: false
+    mgr:
+      disabled: false
+    osd:
+      disabled: false
+```
+
+The probe itself can also be overridden, refer to the [Kubernetes documentation](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#define-a-liveness-command).
```

Comment:
Why do we need to expose overriding the liveness probe? I thought we were just exposing whether to disable it or to specify the interval. I wouldn't expect anyone to need to change the definition of the liveness probe.

Reply:
The idea is to be able to adjust some of the values from the Probe, I don't think it would be used to override the whole Probe but that's the easiest way. Otherwise, we would need to expose other settings from the Probe and do an accurate matching...